crates/uv-resolver/src/resolution/graph.rs

```diff
@@ -39,233 +38,233 @@ impl ResolutionGraph {
     /// Create a new graph from the resolved PubGrub state.
     #[allow(clippy::too_many_arguments)]
     pub(crate) fn from_state(
-        selection: &SelectedDependencies<UvDependencyProvider>,
-        pins: &FilePins,
-        packages: &FxOnceMap<PackageName, Arc<VersionsResponse>>,
-        distributions: &FxOnceMap<VersionId, Arc<MetadataResponse>>,
-        state: &State<UvDependencyProvider>,
+        index: &InMemoryIndex,
         preferences: &Preferences,
+        resolution: Resolution,
     ) -> anyhow::Result<Self, ResolveError> {
+        // Collect all marker expressions from relevant pubgrub packages.
+        let mut markers: FxHashMap<(PackageName, Version), MarkerTree> = FxHashMap::default();
```

Comment:
This might need to be keyed on `(PackageName, Option<ExtraName>)`?

Reply:
Yup, nice catch. Fixed!